petitionemail.php

```diff
@@ -163,3 +173,30 @@ function petitionemail_civicrm_postProcess($formName, &$form) {
       break;
   }
 }
+
+/**
+ * Implements hook_civicrm_fieldOptions().
+ *
+ * Add all the groups listed in allowedgroups_for_eventdedupe to the duplicate_if_in_groups custom field
+ * which is used to select a group that is used in hook_civicrm_findDuplicates to choose whether to duplicate
+ * or merge contact on event registration.
+ */
+function petitionemail_civicrm_fieldOptions($entity, $field, &$options, $params) {
+  if ($entity == 'Survey' && ($field == 'custom_427')) {
```

Comment:
It looks like a custom field id slipped in here. I think this would just need a look up to find out which custom field id matches the field with the name MessageTemplate. Otherwise this looks great. Thanks so much for submitting it!

Reply:
Oops! Now resolved